__tests__/roundtrip_test.ml

```diff
@@ -13,6 +11,40 @@ let run_test ~name ~write ~read ~data =
     |> toEqual data'
   )
 
+let run_read_test ~name ~read ~data ~expected =
+  let open Expect in
+  let decode = Atdgen_codec_runtime.Decode.decode read in
+  let data' = decode data in
+  test name (fun () ->
+    expect data'
+    |> toEqual expected
+  )
+
+
+(** Test that decoding data that wouldn't normally be encoded still equals the correct value *)
```

Comment:
I call it "data that wouldn't normally be encoded" not because it isn't JSON, but as you inferred because it's `null` when the encoder will only produce `undefined`. I'll clarify the comment and the test name.

You're also right, it doesn't belong in a "round trip test". I was just being lazy and keeping all the tests in one file. I'll fix that too.